### PR TITLE
Blog App: Create a data model

### PR DESCRIPTION
**Pull request by Timbar09**
In this PR, I created a new Rails app and a data mode for the app.

**Changes**

- Set up the Rails app with a Postgres database.
- Created `User`, `Post`, `Comment`, and `Like` models.
- Created necessary Migration files.
- Added indexes to corresponding foreign key columns.

### CODE REVIEW

File: db/migrate/20230503083736_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :user, null: false, foreign_key: true, index: true
```
Comment (pinkmoon25) [inline db/migrate/20230503083736_create_posts.rb]: - [x] Good job setting up data models, however, please notice as mentioned in the requirements 
Screenshot from 2023-05-03 16-42-42

we have to make sure the column name for referencing user should be author_id you can do that with `t.references :author, foreign_key: { to_table: 'users' }`

notice: you will have to create a new migration file for this

File: db/migrate/20230503083828_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :user, null: false, foreign_key: true, index: true
```
Comment (pinkmoon25) [inline db/migrate/20230503083828_create_comments.rb]: - [x] Same thing here, this column should also be author_id, If you see the ERD diagram you will notice that the column name for referencing user is named author_id

File: db/migrate/20230503083903_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true, index: true
```
Comment (pinkmoon25) [inline db/migrate/20230503083903_create_likes.rb]: - [x] Same as above here also it should be author_id

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @Timbar09 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Rubocop checks :heavy_check_mark: 
gitfow :heavy_check_mark: 
Good PR and commits :heavy_check_mark: 
Postgresql adapter :heavy_check_mark: 
data models :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Timbar09 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thanks, @TadesseDev. 🙏

**Pull request merged**